### DISCUSSION TITLE
Automate submatrix command

### DISCUSSION THREAD

**Discussion by Nacho-capparelli** [wxMaxima-developers/wxmaxima#1972] [Q&A]
Cheers from argentina
I use maxima-wxmaxima for teaching at university and working. I teach on matrix method for mechanical structures (FEM or finite elementos method). Im looking to automate submatrix comand, but can't find a way to add índices to extract rows and columns in a automatic way. I have a list of índices to extract but don't find a way to autowrite the numbers with comma without brackets inside the command. 

submatrix(1,2,11,12,AA,1,2,11,12);

AA: being the original matrix

I get the automated list [1,2,11,12] but can't deposit it in the command submatrix() without rectangular brackets. Is there a way to make this??? 

Tried, print, concat, some string commands but can't reach the objective, could You please help me? I'm very thankful!!! 
                           Ignacio Capparelli

**Comment by Observatorio-de-Matematica** [comment]
@Nacho-capparelli

I left you a response in the Maxima mailing list.

**Reply by daute** [reply]
Something like that?

```
(%i1) AA:matrix([1,2,3,4],[5,6,7,8],[9,10,11,12]);
                               [ 1  2   3   4  ]
                               [               ]
(%o1)                          [ 5  6   7   8  ]
                               [               ]
                               [ 9  10  11  12 ]
(%i2) l:[2,3];
(%o2)                               [2, 3]
(%i3) s:sconcat(printf(false, "submatrix(~{~a~^, ~}, ", l),"AA, ",printf (false, "~{~a~^, ~})", l));
(%o3)                      submatrix(2, 3, AA ,2, 3);
(%i4) eval_string(s);
(%o4)                              [ 1  4 ]
```

The `printf()` statements convert the (Maxima) list to a list of comma separated values, then I `concat()` the string, which I later evaluate with `eval_string()`.

Best regards, Wolfgang
